### PR TITLE
wifi-2649 Fix Captive portal vifC vifS mismatch

### PR DESCRIPTION
**Pull request by NagConnectus**
Mismatch in vifC and vifS for captive portal is
leading to opennds restarts and memory leakage.
Fix by making sure the vifC and vifS are in parity.

uci_blob not being freed, fix by making uci_blob
as global so that it need not be freed and can
be reused instead.

Signed-off-by: Chaitanya Godavarthi <chaitanya.kiran@netexperience.com>

Cleaning up captive portal code to improve stability and reduce memory leaks

Signed-off-by: Nagendrababu <nagendrababu.bonkuri@connectus.ai>

### CODE REVIEW

**Review by yashvardhan-netex — COMMENTED** [review]
Please add details of the tests performed to the defect in Jira.

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/captive.c [hunk 1]
```diff
@@ -403,42 +415,43 @@ void splash_page_logo(char* dest_file, char* src_url)
...
 
 void captive_portal_get_current_urls(char *ifname, char *splash_logo, char *back_image, char *user_file)
 {
 	char *buf = NULL;
 	struct blob_attr *tc[__NDS_ATTR_MAX] = { };
-	struct blob_buf url_buf={ };
-	struct uci_section *cp_section;
+	struct uci_section *cp_section = NULL;
+	struct uci_package *opennds = NULL;
+	struct uci_context *caps_uci = NULL;
```
[the remark this answers is not in the document]
Comment (chaitanyagodavarthi) [reply feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/captive.c:443]: done

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/fixup.c [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+                return NULL;
+
+        memset(vif, 0, sizeof(*vif));
+        strncpy(vif->name, ifname, IF_NAMESIZE);
+        vif->avl.key = vif->name;
+        avl_insert(&vif_fixup_tree, &vif->avl);
+        return vif;
+}
+
+void vif_fixup_del(char *ifname)
```
[the remark this answers is not in the document]
Comment (chaitanyagodavarthi) [reply feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/fixup.c:47]: I don't see a way to delete the interface from the list, We do set/reset the captive portal flag in the list of interfaces.

[the remark this answers is not in the document]
Comment (yashvardhan-netex) [reply feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/fixup.c:47]: May be inside target_vif_config_del, when the vif is deleted, you would want to delete the entry from this list.

**Review by yashvardhan-netex — APPROVED** [review]

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/utils.c [hunk 3]
```diff
@@ -512,6 +512,20 @@ bool vif_state_to_conf(struct schema_Wifi_VIF_State *vstate,
 	}
 	vconf->custom_options_len = vstate->custom_options_len;
 
+
```
Comment (RickSommerville) [inline feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/utils.c:515]: Whitespace extra ?

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/vif.c [hunk 4]
```diff
@@ -1647,10 +1647,12 @@ static int ap_vif_config_set(const struct schema_Wifi_Radio_Config *rconf,
 	if (changed->captive_portal)
 		vif_captive_portal_set(vconf,(char*)vconf->if_name);
 
+
```
Comment (RickSommerville) [inline feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/vif.c:1650]: Whitespace

**Pull request closed without merging**